explore_costmap:
  explore_costmap:
    ros__parameters:
      footprint: "[ [0.48, 0.38], [0.48, -0.38], [-0.48, -0.38], [ -0.48, 0.38 ] ]"
      update_frequency: 1.0
      publish_frequency: 1.0
      global_frame: map
      robot_base_frame: base_link
      use_sim_time: True
      # robot_radius: 0.22
      resolution: 0.05
      track_unknown_space: true
      plugins: ["static_layer", "obstacle_layer", "bounded_explore_layer"]
      # plugins: ["static_layer", "obstacle_layer", "inflation_layer"]
      obstacle_layer:
        plugin: "nav2_costmap_2d::ObstacleLayer"
        enabled: True
        observation_sources: scan
        scan:
          topic: /camera/points
          max_obstacle_height: 2.0
          clearing: True
          marking: True
          data_type: "PointCloud2"
          raytrace_max_range: 3.0
          raytrace_min_range: 0.0
          obstacle_max_range: 2.5
          obstacle_min_range: 0.0
          max_obstacle_height: 0.4
      static_layer:
        plugin: "nav2_costmap_2d::StaticLayer"
        map_subscribe_transient_local: True
      bounded_explore_layer:
        plugin: "frontier_exploration/BoundedExploreLayer"
        enabled: true
        resize_to_boundary: false
        frontier_travel_point: middle
        explore_clear_space: false
      always_send_full_costmap: True

explore_server:
  ros__parameters:
    frequency: 1.0
    goal_aliasing: 2.0